pkg/rbd/rbd.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// List retrieves and displays the Ceph block pools and their associated images.
+func ListImages(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace string, clusterNamespace string) {
+	// List Ceph Block Pools
+	blockPoolList, err := clientsets.Rook.CephV1().CephBlockPools(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("failed to list CephBlockPools: %w", err))
+	}
+
+	// Initialize map to store block pool names and associated Rados namespaces
+	blockPoolNames := make(map[string][]string)
+	for _, blockPool := range blockPoolList.Items {
+		blockPoolNames[blockPool.Name] = []string{"--", "--"}
+	}
+
+	// List Ceph Block Pool Rados Namespaces
+	blockPoolNamespaceList, err := clientsets.Rook.CephV1().CephBlockPoolRadosNamespaces(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("failed to list CephBlockPoolRadosNamespaces: %w", err))
+	}
+	for _, blockPoolNameSpace := range blockPoolNamespaceList.Items {
+		var name string
+		// Check if Spec.BlockPoolName exists; otherwise, use ObjectMeta.Name
+		if blockPoolNameSpace.Spec.Name != "" {
+			name = blockPoolNameSpace.Spec.Name
+		} else {
+			name = blockPoolNameSpace.ObjectMeta.Name
+		}
+		if _, exists := blockPoolNames[blockPoolNameSpace.Spec.BlockPoolName]; exists {
+			blockPoolNames[blockPoolNameSpace.Spec.BlockPoolName][1] = name
```

Comment:
This looks wrong, we need to keep on appending the namespaces as a single blockpool can have multiple namespaces?

Reply:
done